### PR TITLE
replicate 4chan-x randomized file name

### PR DESCRIPTION
**Pull request by benyben27**

### CODE REVIEW

**Comment by moffatman** [comment]
Seems OK, can you give an example of one of the filenames? It just looks like a random timestamp I guess?

File: lib/widgets/reply_box.dart [hunk 1]
```diff
@@ -822,8 +822,8 @@ Future<void> _handleImagePaste({bool manual = true}) async {
 				overrideAttachmentFilename = '${_filenameController.text.normalizeSymbols}.${attachmentExt!}';
 			}
 			if (settings.randomizeFilenames && attachment != null) {
-				const alphanumericCharacters = 'AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTtUuVvWwXxYyZz1234567890';
-				overrideAttachmentFilename = '${List.generate(12, (i) => alphanumericCharacters[random.nextInt(alphanumericCharacters.length)]).join('')}.${attachmentExt!}';
+				const temp = 24 * 60 * 60 * 1000 * 365 * 1000;
+				overrideAttachmentFilename = (DateTime.now().millisecondsSinceEpoch * 1000 - (random.nextDouble() * temp).floor()).toString() + '.' + attachmentExt!;
```
Comment (moffatman) [inline lib/widgets/reply_box.dart:826]:
```dart
overrideAttachmentFilename = '${DateTime.now().subtract(const Duration(days: 365) * random.nextDouble()).microsecondsSinceEpoch}.${attachmentExt!}';
```

**Comment by benyben27** [comment]
also this is how 4chan-x creates a random file name https://github.com/ccd0/4chan-x/blob/103c9849cc8aeba4b1521dad5176ac31d9ba69d3/src/Posting/QR.post.coffee#L210
the point of replicating it is to not stand out

**Comment by moffatman** [comment]
Sure, I see the reason. Can you change the code as I put in the annotation, it matches dart style better and does the exact same calculation but is clearer to understand imo. I guess if you always want it to end in 000, you can use `.millisecondsSinceEpoch * 1000` instead of `.microsecondsSinceEpoch`.

**Comment by benyben27** [comment]
should've tested it before committing it 
but behavior isnt the same 
it didnt end with 000 before

**Comment by moffatman** [comment]
Thx 👍

**Comment by benyben27** [comment]
ok i think the issue that im seeing is that i tested in on dartpad, instead of doing it properly 
because it always returns 000 at the end on microsecondsSinceEpoch, hopefully it works in the app or on your side

**Comment by moffatman** [comment]
In Dart's JavaScript runtime I think DateTime is limited millisecond precision. I checked and it works properly in the app

**Pull request merged**